commons/utils/src/main/java/in/projecteka/jataayu/util/repository/PreferenceRepository.kt

```diff
@@ -0,0 +1,75 @@
+package `in`.projecteka.jataayu.util.repository
+
+import android.content.SharedPreferences
+import androidx.core.content.edit
+
+interface PreferenceRepository {
+
+    var name: String?
+
+    var mobileIdentifier: String?
+
+    var pinCreated: Boolean
+
+    var isUserLoggedIn: Boolean
+
+    var isUserRegistered: Boolean
+
+    var isUserAccountCreated: Boolean
+
+    var hasProviders: Boolean
+
+    fun resetCredentials()
+}
+
+class PreferenceRepositoryImpl(private val sharedPreferences: SharedPreferences) :
+    PreferenceRepository {
+
+    companion object {
```

Comment:
create key as enum. so you can access all the keys by one shot. It will be easy to remove. when removing you can iterate through key and call remove method.  If any new key added you no need to bother about removing that key from shared preference. enum do that job.

Reply:
the benefit of using an enum is lost (using the entrySet to iterate) because first, SharedPreferences require a String Key, with an enum that would mean overriding the constructor for that purpose and having a string value, losing the value of enumerations since now they'd be complex objects.  Additionally, not all values in the shared preferences are of the same type, which means, during an iteration I would have to write a if/else tree to delete/modify the value. If i had to delete all the shared preference values, I would just delete the file in that case.